### DISCUSSION TITLE
question about prometheus metrics in customer connector

### DISCUSSION THREAD

**Discussion by cstmgl** [strimzi/strimzi-kafka-operator#7053] [Q&A]
Hi all,

Sorry if my question is really basic but I'm a bit confused, mainly because I have little experience with Prometheus.
When using the Kafka Connector framework/API I can expose already some metrics to Prometheus and that works fine.
Now I would like to build my custom connector and build some custom metrics.
My question is how could I expose my metrics from my application? 
Do I need to create my custom metrics service? And how could Prometheus point also to it?

Or would the framework be smart enough to know how to expose the metrics?

Is there any example of some open source connectors that expose metrics I could look at that people are aware of?
I think I know how to collect and generate the metrics I just don't understand how they would be exposed to Prometheus without having a service on the application.
And if the framework already exposes that service how can I delegate my application into that service.

Thanks for any feedback, sorry if the question is not clear.

**Comment by scholzj** [comment]
The way Strimzi provides the Prometheus metrics from Kafka Connect is using the Prometheus JMX Exporter. It runs as an agent, and gets the Kafka metrics through JMX. The ConfigMap which is used to configure the metrics. It tells the JMX Exporter which JMX beans should be exposed as Prometheus metrics and how. So your connector need to plug into the KAfka metrics to plug into the ecosystem instead of just creating its own Prometheus export. But TBH, how exactly to plug it is question more on Kafka than Strimzi. Checking how some existing connectors exposing metrics do it might be also an option.
